cmd/arc/services/api.go

```diff
@@ -58,22 +58,19 @@ func StartAPIServer(logger *slog.Logger, arcConfig *config.ArcConfig) (func(), e
 
 	shutdownFns := make([]func(), 0)
 
-	tracingEnabled := false
-	if arcConfig.Tracing != nil && arcConfig.Tracing.DialAddr != "" {
+	if arcConfig.Tracing != nil && arcConfig.Tracing.DialAddr != "" && arcConfig.Tracing.Enabled {
```

Comment:
In some places there's a check
```
arcConfig.Tracing != nil && arcConfig.Tracing.DialAddr != "" && arcConfig.Tracing.Enabled
```
and in others there's only
```
arcConfig.Tracing != nil && arcConfig.Tracing.Enabled
```

I'm thinking whether it would make sense to create a helper function, e.g. in `config/utils.go`
```
func isTracingEnabled(tracing *config.TracingConfig) bool {
    return tracing != nil && tracing.DialAddr != "" && tracing.Enabled
}
```
or maybe even this way:
```
func isTracingEnabled(tracing *config.TracingConfig) (bool, error) {
    if tracing == nil {
        return false, nil
    }

    if tracing.Enabled && tracing.DialAddr == "" {
        return nil, errors.New("tracing enabled but dial address was not given")
    }

    return tracing.Enabled
}
```
because if someone wanted to enable tracing but forgot to put a DialAddr we could just stop Arc before running and throw this error.

WDYT?

Reply:
Hey @kuba-4chain 
Yes, good input 👍 

I'll create such a helper function